constellations/gemini-admin/src/emails/accountInfo.js

```diff
@@ -0,0 +1,134 @@
+"use strict";
+require("../account/accountInfo.sass");
+import React from "react";
+import { Link } from "react-router-dom";
+import { getFullName } from "../utils/userUtils.js";
+
+export class AccountInfo extends React.Component {
+    state = {
+        id: 0,
+        email: "",
+        users: [],
+        transactions: [],
+        selectedUserEmails: [],
+        active: false,
+    };
+
+    formatCurrency = (amount) => {
+        return new Intl.NumberFormat("en-US", {
+            style: "currency",
+            currency: "USD",
+        }).format(amount);
+    };
+
+    onCheckUser = (e, userId) => {
+        const users = this.props.users;
+        if (e.target.checked) {
+            users.map((user, index) => {
+                if (user.id == userId) {
+                    var email = this.state.selectedUserEmails;
+                    email.push(user.email);
+                    this.setState({
+                        selectedUserEmails: email,
+                    });
+                }
+            });
+        } else {
+            users.map((user, index) => {
+                if (user.id == userId) {
+                    var email = this.state.selectedUserEmails;
+                    email.splice(email.indexOf(user.email), 1);
+                    this.setState({
+                        selectedUserEmails: email,
+                    });
+                }
+            });
+        }
+    };
+
+    render = () => {
+        const id = this.props.id;
+        const email = this.props.email;
+        const users = this.props.users;
+        const transactions = this.props.transactions;
+        const name = this.props.name;
+
+        var emails = [];
+        users.map((user, index) => {
+            emails.push(user.email);
+        });
+
+        const userRows = users.map((user, index) => {
+            var status = user.isGuardian ? "(guardian" : "(student";
+            status += user.email == email ? ", primary contact)" : ")";
+
+            if (this.state.selectedUserEmails.length > 0) {
+                this.state.selectedUserEmails.map((existingEmail, index) => {
+                    if (emails.indexOf(existingEmail) == -1) {
+                        this.setState({
+                            selectedUserEmails: [],
+                        });
+                    }
+                });
+            }
+            return (
+                <div className="row" key={index}>
+                    <input
+                        type="checkbox"
+                        onChange={(e) => this.onCheckUser(e, user.id)}
+                    />
+                    <span className="column">{getFullName(user)}</span>
+                    <span className="column">{user.email}</span>
+                    <span className="column status">{status}</span>
+                </div>
+            );
+        });
+
+        var balance = 0;
+        transactions.map((transaction, index) => {
```

Comment:
this is a good place for `.forEach` instead of map. Map should be used when you want to create a new list from an original list. With forEach, you simply loop through all the elements and apply a function to each element.

At the end of the day, it's the same thing, but from a semantics perspective, it's more accurate.